src/components/SignTransactionWallets/Web3.tsx

```diff
@@ -107,61 +115,110 @@ export default function SignTransactionWeb3({
-            </div>
-          )}
-        </div>
-        <div className="SignTransactionWeb3-description">{translateRaw('SIGN_TX_EXPLANATION')}</div>
-        <div className="SignTransactionWeb3-footer">
-          <BusyBottom
-            type={
-              walletConfig.id === WalletId.METAMASK
-                ? BusyBottomConfig.METAMASK_SIGN
-                : BusyBottomConfig.GENERIC_WEB3
-            }
-          />
-        </div>
-      </div>
-    </>
+    <SignTransactionWeb3UI
+      walletConfig={walletConfig}
+      walletState={walletState}
+      networkName={networkName}
+      senderAccount={senderAccount}
+    />
   );
 }
+
+const Footer = styled.div`
+  width: 100%;
+  margin-top: 2em;
+`;
+
+const Web3ImgContainer = styled.div`
+  padding-bottom: 3em;
+  display: flex;
+  justify-content: center;
+  align-content: center;
+
+  @media (max-width: ${BREAK_POINTS.SCREEN_SM}) {
+    padding-bottom: 1em;
+  }
+`;
+
+const Web3Img = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  height: 150px;
+  & img {
+    width: 150px;
+  }
+`;
+
+export interface UIProps {
+  walletConfig: IWalletConfig;
+  walletState: WalletSigningState;
+  networkName: string;
+  senderAccount: StoreAccount;
+}
+
+export const SignTransactionWeb3UI = ({
+  walletConfig,
+  walletState,
+  networkName,
+  senderAccount
+}: UIProps) => (
+  <Box p="2.5em">
+    <Heading fontSize="32px" textAlign="center" fontWeight="bold">
+      {translate('SIGN_TX_TITLE', {
+        $walletName: walletConfig.name || WALLETS_CONFIG.WEB3.name
+      })}
+    </Heading>
+    <Body textAlign="center" fontSize="2" mb={SPACING.BASE} paddingTop="16px">
+      {translate('SIGN_TX_WEB3_PROMPT', {
+        $walletName: walletConfig.name || WALLETS_CONFIG.WEB3.name
+      })}
+    </Body>
+    <Web3ImgContainer>
+      <Web3Img>
+        <img src={walletConfig.icon} />
+      </Web3Img>
+    </Web3ImgContainer>
+
+    <>
+      <Box variant="columnCenter">
+        {walletState === WalletSigningState.REJECTED && (
+          <InlineMessage textAlign="center">{translate('SIGN_TX_WEB3_REJECTED')}</InlineMessage>
+        )}
+        {walletState === WalletSigningState.NETWORK_MISMATCH && (
+          <InlineMessage textAlign="center">
+            {translate('SIGN_TX_WEB3_FAILED_NETWORK', {
+              $walletName: walletConfig.name,
+              $networkName: networkName
+            })}
+          </InlineMessage>
+        )}
+        {walletState === WalletSigningState.ADDRESS_MISMATCH && (
+          <InlineMessage textAlign="center">
+            {translate('SIGN_TX_WEB3_FAILED_ACCOUNT', {
+              $walletName: walletConfig.name,
+              $address: senderAccount.address
+            })}
+          </InlineMessage>
+        )}
+        {walletState === WalletSigningState.SUBMITTING && (
+          <InlineMessage textAlign="center" type={InlineMessageType.INFO_CIRCLE}>
```

Comment:
I think we should use another color for this notice, it looks like a warning.
![image](https://user-images.githubusercontent.com/1561200/106126090-3a209a00-615d-11eb-8a25-6ce856ddf301.png)

Reply:
We use this color for inline messaging in Send Assets too. Would need some feedback from design to change that
![image](https://user-images.githubusercontent.com/29407814/106190817-05154700-615f-11eb-8ee9-0ea6f8bff413.png)

Reply:
Yes. For warnings not for info

Reply:
![image](https://user-images.githubusercontent.com/29407814/106680555-ab07ed80-6573-11eb-8354-05e768cc16f8.png)
dis betta color?

Reply:
I think it is fine to use white unless we have a design decision to use another color.

Reply:
Changed to purple as per taylor